Polyhedron/demo/Polyhedron/Viewer.cpp

```diff
@@ -65,11 +71,14 @@ Viewer::Viewer(QWidget* parent, bool antialiasing)
   pickMatrix_[15]=1;
   prev_radius = sceneRadius();
   axis_are_displayed = true;
+  has_text = false;
+  i_is_pressed = false;
 }
 
 Viewer::~Viewer()
 {
   delete d;
+  delete painter;
 }
```

Comment:
All data members of `Viewer` should be hidden in the `d` pointer. That avoids to expose implementation details in `Viewer.h`.

For explanations of d-pointers, see:
- https://wiki.qt.io/D-Pointer
- https://en.wikipedia.org/wiki/Opaque_pointer

That is a common implementation trick, that I discovered first in the source code of Qt4.